cdm/src/main/java/thredds/crawlabledataset/s3/CrawlableDatasetAmazonS3.java

```diff
@@ -176,24 +143,10 @@ public boolean isCollection() {
      */
     @Override
     public long length() {
-        // If the summary is already in the cache, return it.
-        // It'll have been added by a listDatasets() call on the parent directory.
-        S3ObjectSummary objectSummary = objectSummaryCache.getIfPresent(s3uri);
-        if (objectSummary != null) {
-            return objectSummary.getSize();
-        }
+        ThreddsS3Metadata metadata = threddsS3Client.getMetadata(s3uri);
 
-        /* Get the metadata directly from S3. This will be expensive.
-         * We get punished hard if length() and/or lastModified() is called on a bunch of datasets without
-         * listDatasets() first being called on their parent directory.
-         *
-         * So, is the right thing to do here "getParentDataset().listDatasets()" and then query the cache again?
-         * Perhaps, but listDatasets() throws an IOException, and length() and lastModified() do not.
-         * We would have to change their signatures and the upstream client code to make it work.
-         */
-        ObjectMetadata metadata = threddsS3Client.getObjectMetadata(s3uri);
-        if (metadata != null) {
-            return metadata.getContentLength();
+        if (metadata instanceof ThreddsS3Object) {
```

Comment:
Well, I did have an isCollection method on the metadata object at one point - but I though just checking the type of object was better - seemed wrong having an isCollection method on an S3Object.